File: backend/src/routes/events.routes.ts
import { Router } from 'express'
import { requireAuth } from '../middlewares/auth.middleware'
import { listarEventos, criarEvento, editarEvento, deletarEvento } from '../controllers/events.controller'

const router = Router()

// Todas as rotas de eventos exigem autenticação
router.use(requireAuth)

router.get('/',     listarEventos)
router.post('/',    criarEvento)
router.put('/:id',  editarEvento)
router.delete('/:id', deletarEvento)

export default router

File: backend/internal/events/events_routes.go
package events

import (
	"github.com/gin-gonic/gin"
	"github.com/jackc/pgx/v5/pgxpool"

	"beeplanerofc/backend-go/internal/middleware"
	"beeplanerofc/backend-go/internal/supabaseauth"
)

// RegisterRoutes registra as rotas de eventos no grupo /api/events. Todas exigem login.
func RegisterRoutes(router *gin.RouterGroup, db *pgxpool.Pool, authClient *supabaseauth.Client) {
	ctrl := NewController(db)

	router.Use(middleware.RequireAuth(db, authClient))

	router.GET("", ctrl.ListarEventos)
	router.POST("", ctrl.CriarEvento)
	router.PUT("/:id", ctrl.EditarEvento)
	router.DELETE("/:id", ctrl.DeletarEvento)
}
